plutus-tx-plugin/test/IsData/Spec.hs

```diff
@@ -67,6 +70,23 @@ isDataRoundtrip a =
         unsafeRoundtrip = IsData.unsafeFromBuiltinData d P.== a
     in safeRoundtrip && unsafeRoundtrip
 
+AsData.asData [d|
+  data SecretlyData = FirstC () | SecondC Integer
+  |]
+
+AsData.asData [d|
+  data RecordConstructor a = RecordConstructor { x :: a, y :: Integer }
+  |]
```

Comment:
I'd be tempted to add a test case converting between these two types, plus equality comparison, say

```haskell
dataToData :: CompiledCode (RecordConstructor Integer -> SecretlyData)
dataToData = plc (Proxy @"dataToData")
  (\(x@(RecordConstructor a b)) ->
    if x == RecordConstructor 3 4
      then SecondC (Builtins.addInteger a b)
      else FirstC ())
```

Reply:
Yes, good idea.